app/scripts/directives/simple-slider.js

```diff
@@ -1,20 +1,50 @@
 angular.module('angularSimpleSlider')
-  .directive('simpleSlider', ['SimpleSliderService', function (SimpleSliderService) {
+  .directive('simpleSlider', ['SimpleSliderService', '$timeout', function (SimpleSliderService, $timeout) {
 
     'use strict';
 
     return {
 
-      restrict: 'E',
-      scope: {},
+      restrict: 'AE',
+      scope: {
+        onChange: '&',
+        slide: '=?simpleSliderSlide',
+        slider: '=?simpleSlider'
```

Comment:
I have to confess that my angular skills are a bit rusty these days, so help me out with this one.

Is it really a good idea to expose the `SimpleSlider` object as a `simple-slider` attribute?

My original idea was to hide everything behind the directive implementation and the consumer would just play with the attributes. I don't really know what are the trade-offs involved here. Since you're already exposing `onchange` and the `simple-slider-slide` attribute it seems that it would make sense to keep the consumers away from the simpleSlider instance, have you give it some thought? I would love to have some feedback :blush: